util/oidc/oidc.go

```diff
@@ -43,6 +43,21 @@ const (
 	AccessTokenCachePrefix      = "access_token"
 )
 
+// Request Metadata for warning message
+type RequestMetadata struct {
+	ipAddr string
+}
+
+func (r *RequestMetadata) String() string {
+	return fmt.Sprintf("Client IP: %s", r.ipAddr)
+}
+
+func newRequestMetadata(r *http.Request) *RequestMetadata {
+	return &RequestMetadata{
+		ipAddr: r.RemoteAddr,
+	}
+}
+
```

Comment:
Usually metadata is good to add as structured data, e.g. https://github.com/argoproj/argo-cd/blob/32cc6638f78e15652c24eb2cd6403e6e7ec26ab0/controller/appcontroller.go#L101-L108

Reply:
agree. adjusting